### PR TITLE
create decorators for cli methods

### PR DESCRIPTION
**Pull request by rmudambi**

### CODE REVIEW

File: src/covid_shared/cli_tools/decorators.py [hunk 1]
```diff
@@ -48,3 +50,109 @@ def _wrapped(*args, **kwargs):
...
+            default=R_SINGULARITY_IMAGE_PATH,
+            help=(f"Path to a specific R singularity. Can pass 'None' to use active conda environment. "
+                  f"By default uses {R_SINGULARITY_IMAGE_PATH}.")
+        )(entry_point)
+
+        return entry_point
+    return wrapper
+
+
+def add_create_directory_options(default_output_root: Path):
```
Comment (collijk) [inline src/covid_shared/cli_tools/decorators.py:72]: Probably call this `add_output_options`

File: src/covid_shared/cli_tools/decorators.py [hunk 2]
```diff
@@ -48,3 +50,109 @@ def _wrapped(*args, **kwargs):
...
+        entry_point = click.option(
+            '-p', '--production-tag',
+            type=click.STRING,
+            help='Tags this run as a production run.'
+        )(entry_point)
+
+        entry_point = click.option(
+            '-b', '--mark-best',
+            is_flag=True,
+            default=False,
```
Comment (collijk) [inline src/covid_shared/cli_tools/decorators.py:95]: don't need the default arg.

File: src/covid_shared/cli_tools/decorators.py [hunk 3]
```diff
@@ -48,3 +50,109 @@ def _wrapped(*args, **kwargs):
...
+    Decorator to set CLI command options required when producing outputs.
+    """
+    def wrapper(entry_point: types.FunctionType):
+
+        entry_point = click.option(
+            "-o", "--output-root",
+            type=click.Path(file_okay=False),
+            default=default_output_root,
+            help=(f"Directory to outputs. Defaults to {default_output_root}/YYYY_MM_DD.VV for today's date and the "
+                  f"newest uncreated version")
```
Comment (miker985) [inline src/covid_shared/cli_tools/decorators.py:83]: my linter complains about f-strings that don't do any substitution.

Comment (collijk) [reply src/covid_shared/cli_tools/decorators.py:83]: I'm super sloppy about this.  But it is right.

File: src/covid_shared/cli_tools/decorators.py [hunk 4]
```diff
@@ -48,3 +50,109 @@ def _wrapped(*args, **kwargs):
...
+    ----------
+    data_source
+        The name of the data source. Multiple words should be separated by '-'
+    default
+        The default location of the dependency. If None is provided, there is no default and the option is required.
+    """
+    def wrapper(entry_point: types.FunctionType):
+        click_args = {
+            'type': click.Path(file_okay=False),
+            'help': f'Version of the {data_source.replace("-", " ")} to use. Defaults to "{default}"',
```
Comment (miker985) [inline src/covid_shared/cli_tools/decorators.py:117]: `help` text incomplete - if default is None there actually is no default and it's mandatory. Maybe set `help` situationally in the below if/else block?

Comment (rmudambi) [reply src/covid_shared/cli_tools/decorators.py:116]: Good call.

**Pull request merged**